### PR TITLE
Add type caster for string_view.

### PR DESCRIPTION
**Pull request by qnzhou**
A small update to automatically type cast `std::string_view`.

### CODE REVIEW

**Comment by wjakob** [comment]
Thanks @qnzhou — this looks great, but it is missing a testcase. Could you add one for both directions of the conversion?

File: tests/test_stl.py [hunk 1]
```diff
@@ -326,3 +326,18 @@ def test32_list():
     assert t.identity_list([1, 2, 3]) == [1, 2, 3]
     assert t.identity_list(()) == []
     assert t.identity_list((1, 2, 3)) == [1, 2, 3]
+
+def test33_string_and_string_view():
+    assert t.identity_string("") == ""
+    assert t.identity_string("orange") == "orange"
+    assert t.identity_string("橘子") == "橘子"
+    assert t.identity_string("ส้ม") == "ส้ม"
+    assert t.identity_string("البرتقالي") == "البرتقالي"
```
Comment (wjakob) [inline tests/test_stl.py:335]: Is there something funny going on with parentheses here?

File: tests/test_stl.py [hunk 2]
```diff
@@ -326,3 +326,18 @@ def test32_list():
...
+    assert t.identity_string("🍊") == "🍊"
+
+    assert t.identity_string_view("") == ""
+    assert t.identity_string_view("orange") == "orange"
+    assert t.identity_string_view("橘子") == "橘子"
+    assert t.identity_string_view("ส้ม") == "ส้ม"
+    assert t.identity_string_view("البرتقالي") == "البرتقالي"
```
Comment (wjakob) [inline tests/test_stl.py:342]: ditto

File: tests/test_stl.py (hunk 1 above)
Comment (qnzhou) [reply tests/test_stl.py:335]: I don't quite understand. Is there anything wrong with the parentheses?

Comment (wjakob) [reply tests/test_stl.py:335]: Could it be a rendering issue? At least on the GitHub web interface, I don't see a closing perenthesis after the arabic strings.

**Comment by wjakob** [comment]
This is what it looks like on my screen. Screen Shot 2022-09-28 at 17 34 21

File: tests/test_stl.py (hunk 1 above)
Comment (qnzhou) [reply tests/test_stl.py:335]: This is what I see:
image

**Pull request merged**